### ISSUE TITLE
VSCode 1.44.2 stops working via proxy

### ISSUE THREAD

**Issue by wkia** [microsoft/vscode#95521]
- VSCode Version: 1.44.2
- OS Version: Windows 7, Windows 10, x64

Steps to Reproduce:

1. You need to work behind a proxy. 
2. Install VSCode 1.43.1 (User)
3. Set up proxy settings in Settings, Http:Proxy e.g. "http://user:password@my.proxy.com:8080"
4. Click Help->Check for Updates... and wait "new version available" notification
5. Click "Restart to update"
6. After restart, make sure if you use 1.44.2 version
7. Also make sure if proxy settings are the same
8. Click Help->Check for Updates... and get "net::ERR_PROXY_CONNECTION_FAILED" error

Does this issue occur when all extensions are disabled?: No extensions installed.

**Comment by Ynewtime** [comment]
Same problem here.

VS Code version 1.44.2
Windows 10 1909 18363.720
Problem: Can not install extensions, and get "net::ERR_PROXY_CONNECTION_FAILED" error, no matter a proxy is set in `settings.json` or not. Behind a proxy, it just won't work normal as usual. Switching to network not required with a proxy works fine.

After uninstalling this version(1.44.2) and clear all the VS Code config files in `~/.vscode` and `User/AppData/Roaming/Code`, I tried installing version 1.45.0 insider, still not working behind a proxy. Uninstall it, then reinstall version 1.44.1, it works behind a proxy! But then it auto updated again... So I have to reinstall it again and set `update.mode = none`...

**Comment by gjsjohnmurray** [comment]
I'm not seeing anything obviously proxy-related in the diffs between 1.44.1 and 1.44.2

https://github.com/microsoft/vscode/compare/1.44.1...1.44.2

**Comment by chrmarti** [comment]
The user setting `http.proxy` doesn't affect extension or VS Code downloads in 1.43 or 1.44. (It will with 1.45.)

@deepak1556 Are you aware of changes between 1.43 and 1.44 that could affect proxy handling?

@Ynewtime Can you confirm it works with 1.44.1? Did you mean 1.43.1?

**Comment by deepak1556** [comment]
There was no updates to the electron runtime between `1.43` and `1.44`. This definitely seems like an error from the user system.

@Ynewtime does the proxy setup work with chrome/edge on your system ? Did you try configuring the proxy at the system level instead of defining `http.proxy` variable ?

**Comment by wkia** [comment]
@deepak1556 Taking into account that there were no changes except VSCode update, and there were no such errors fot all updates before 1.44.2, which kind of an error from the user system do you mean?

**Comment by deepak1556** [comment]
`net::ERR_PROXY_CONNECTION_FAILED` usually refers to an unreachable proxy server. Can you confirm if other applications like chrome/edge are operating with the proxy ?

Can you also launch code with `--log-net-log=<path>/netlog.json` and attach the log here.

**Comment by s-kocher** [comment]
Same problem behind a corporate proxy needing authentication (kerberos but no totally sure about that).
Usually my vscode uses the proxy through http_proxy / https_proxy environment variable and is was working fine (version check / update / extension / ...) until recently (probably because of the last insider version update I installed, i was stuck to 1.45 of 17 april).

I tested to use instead px proxy (https://github.com/genotrance/px) that act as a proxy handling the authentication to the corporate proxy and HTTPS_PROXY=http://localhost:3128 (same for http_proxy variable) and it's working as a workaroung waiting the fix.

VSCode -> read http[s]_proxy env variable -> connect to localhost px proxy -> authenticate to corporate proxy -> internet

My current insider version is based on electron 7.2.2, next version 7.2.3(https://github.com/electron/electron/releases/tag/v7.2.3) is mentioning : `Security: Ensure proxy object is created in the correct context a9bead2`, may be it is the cause...

May be upgrading to electron v7.2.3 for 1.45 is enough instead of waiting bigger electron v8 upgrade (as mentioned in https://github.com/microsoft/vscode/issues/94862)

**Comment by deepak1556** [comment]
@s-kocher can you open a new issue with the relevant details, of when did this issue start, what was the last known working version. 

Also can you launch code with `--log-net-log=netlog.json` , perform some requests (search extensions, update check) and attach the generated log file in the new issue.

`7.2.3` release has nothing to do with any network related operations, please ignore it.

**Comment by inDream** [comment]
I've also faced this issue and latest insider build seems fixed it.
```
Version: 1.45.0-insider
Commit: bd262894bbaac5156d5bf832160b7056c91decea
Date: 2020-04-22T17:16:24.886Z
Electron: 7.2.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.17763
```

**Comment by deepak1556** [comment]
@s-kocher can you try the latest insiders , I think your case should be addressed by https://github.com/microsoft/vscode/issues/94862#issuecomment-617630954

**Comment by Ynewtime** [comment]
> The user setting `http.proxy` doesn't affect extension or VS Code downloads in 1.43 or 1.44. (It will with 1.45.)
> 
> @deepak1556 Are you aware of changes between 1.43 and 1.44 that could affect proxy handling?
> 
> @Ynewtime Can you confirm it works with 1.44.1? Did you mean 1.43.1?

Yes, it works with 1.44.1. I am using it right now(with proxy).

> There was no updates to the electron runtime between `1.43` and `1.44`. This definitely seems like an error from the user system.
> 
> @Ynewtime does the proxy setup work with chrome/edge on your system ? Did you try configuring the proxy at the system level instead of defining `http.proxy` variable ?

Actually, no proxy setup needed to work with chrome/edge, which means no matter I setup a proxy a not, behind my company's proxy, chrome will always work.

---

For more information:

Sorry for the delay. I test 1.44.2 today and it works behind the proxy! Installing extensions and checking for updates, etc., both work now as usual. 

```
Version: 1.44.2 (user setup)
Commit: ff915844119ce9485abfe8aa9076ec76b5300ddd
Date: 2020-04-16T16:36:23.138Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363
```

No proxy config was set though, no matter in `settings.json`, system level proxy or http[s]_proxy environment, I am not sure if this issue was resolved by my company's IT system or not. But now behind proxy, 1.44.1 and 1.44.2 works.

BTW, I have some powershell proxy configs to solve the SSL certification failed issues because my company use self-signed certifications. I don't know if this would help though:

```powershell
[System.Net.WebRequest]::DefaultWebProxy = [System.Net.WebRequest]::GetSystemWebProxy()
[System.Net.WebRequest]::DefaultWebProxy.Credentials = [System.Net.CredentialCache]::DefaultNetworkCredentials
```

And I also install the 1.45.0-insiders to test if it work behind the proxy, unfortunately, it doesn't. Same `net::ERR_PROXY_CONNECTION_FAILED` error. 

```
Version: 1.45.0-insider (user setup)
Commit: bd262894bbaac5156d5bf832160b7056c91decea
Date: 2020-04-22T17:16:24.886Z
Electron: 7.2.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363
```

**Comment by wkia** [comment]
@deepak1556 
After the last @Ynewtime message, I tried to reproduce the issue again and couldn't. VSCode 1.44.2 now works with proxy for me.
So my initial problem is gone now. Thanks!

**Comment by chrmarti** [comment]
@Ynewtime Do you have `HTTP_PROXY`, `HTTPS_PROXY` or `NO_PROXY` environment variables set? Could you retry with todays insiders build (that has a small fix)?

**Comment by Ynewtime** [comment]
> @Ynewtime Do you have `HTTP_PROXY`, `HTTPS_PROXY` or `NO_PROXY` environment variables set? Could you retry with todays insiders build (that has a small fix)?

No, not working, with or without proxy environment variables(i.e. system http_proxy and https_proxy). Same error. I would like to post a screenshot but I am not allowed to upload files without permission, sorry.

```
Version: 1.45.0-insider (user setup)
Commit: 0778354f3a6a2c21b7f738a5e6b02e2f1c765e73
Date: 2020-04-27T05:42:11.567Z
Electron: 7.2.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363
```

**Comment by Ynewtime** [comment]
Log(Window) output:

```
[2020-04-28 08:05:55.477] [renderer1] [error] net::ERR_PROXY_CONNECTION_FAILED: Error: net::ERR_PROXY_CONNECTION_FAILED
    at SimpleURLLoaderWrapper.<anonymous> (electron/js2c/browser_init.js:2623:21)
    at SimpleURLLoaderWrapper.emit (events.js:203:13)
```

Extensions sidebar output:

```
We cannot connect to the Extensions Marketplace at this time, please try again later.
```

**Comment by chrmarti** [comment]
@Ynewtime Can you launch code with `--log-net-log=netlog.json`, perform some requests (search extensions, update check) and attach the generated log file?

**Comment by Ynewtime** [comment]
> @Ynewtime Can you launch code with `--log-net-log=netlog.json`, perform some requests (search extensions, update check) and attach the generated log file?

Problem solved, must be my company's proxy issue. My company provide proxies for different regions, I used to use the HK one and encoutered with the `ERR_PROXY_CONNECTION_FAILED` error with insiders version, but after I switched to different proxy, the error appear no more. So this should be my problem. And thanks for your patience!

**Comment by chrmarti** [comment]
Thanks. Marking as duplicate of #94862 where we solved an issue with 1.44.

**Comment by 7af03dec0b** [comment]
I had the same problem, but something seems to have changed on the Windows side... I used a local script for automatic proxy settings, it worked for a long time, but a few weeks ago I installed an update for Windows and then I had to switch the proxy settings to "Automatically detect settings" and from then on everything works properly, including VS code
